### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "paths[0]" argument must be of type string. Received undefined

### ISSUE THREAD

**Issue by Rock-n-Roll-CRC** [stylelint/stylelint#7166]
### What minimal example or steps are needed to reproduce the bug?
1) Open the project folder;
2) Open the file with .css extension;
3) Save the file;
4) Open the VS Code notification tab;
5) See the error;

### What minimal configuration is needed to reproduce the bug?
```
{
  "extends": "stylelint-config-standard-scss"
}
```

### How did you run Stylelint?
It runs automatically with Visual Studio Code extension

### What did you expect to happen?
No error should produce

### What actually happened?
The following problems were reported:
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined at new NodeError (node:internal/errors:387:5) at validateString (node:internal/validators:121:11) at Object.resolve (node:path:161:9) at Explorer.search (f:\Coding\Test\node_modules\cosmiconfig\dist\Explorer.js:36:45) at getConfigForFile (f:\Coding\Test\node_modules\stylelint\lib\getConfigForFile.js:62:26) at isPathIgnored (f:\Coding\Test\node_modules\stylelint\lib\isPathIgnored.js:24:23) at lintSource (f:\Coding\Test\node_modules\stylelint\lib\lintSource.js:41:26) at Function.standalone [as lint] (f:\Coding\Test\node_modules\stylelint\lib\standalone.js:125:32) at RR.lintDocument (c:\Users\danil\.vscode\extensions\stylelint.vscode-stylelint-1.2.4\dist\start-server.js:71:5596) at process.processTicksAndRejections (node:internal/process/task_queues:96:5)

### Do you have a proposal to fix the bug?
No

**Comment by letsbug** [comment]
cosmiconfig is a stylelint production dependency that was updated to version 8.3 about 2 hours ago，override like this:

```json
{
  "pnpm": {
    "overrides": {
      "cosmiconfig": "~8.2.0"
    }
  }
}
```

**Comment by Rock-n-Roll-CRC** [comment]
> cosmiconfig is a stylelint dependency that was updated to version 8.3 about 2 hours ago，override like this:
> 
> ```json
> {
>   "pnpm": {
>     "overrides": {
>       "cosmiconfig": "~8.2.0"
>     }
>   }
> }
> ```

Should'nt it be updated automatically with npm update? Is it expected to be made manually and not with npm?

**Comment by danielfdickinson** [comment]
The problem is that it automatically updates to 8.3.0 which has the issue. (I just got bitten by this as well).
For me, adding `--config .stylelintrc.json` also works around the issue.

**Comment by Scotchester** [comment]
Related issue opened recently in cosmiconfig repo: https://github.com/cosmiconfig/cosmiconfig/issues/317

**Comment by Scotchester** [comment]
@danielfdickinson Thanks, that's a more elegant workaround than forcing cosmiconfig to a lower version!
